generate/genfile/genfile_test.go

```diff
@@ -536,7 +536,63 @@ stack_description=
 					),
 				},
 			},
-			wantErr: errors.E(genfile.ErrLabelConflict),
+			want: []result{
+				{
+					name:      "test.yml",
+					condition: true,
+					file: genFile{
+						origin: "/stack/test.tm",
+						body:   "test",
+					},
+				},
+				{
+					name:      "test.yml",
+					condition: true,
+					file: genFile{
+						origin: "/stack/test.tm",
+						body:   "test2",
+					},
+				},
+			},
+		},
+		{
+			name:  "same labels but different condition",
+			stack: "/stack",
+			configs: []hclconfig{
+				{
+					path: "/stack/test.tm",
+					add: generateFile(
+						labels("test"),
+						str("content", "test"),
+						boolean("condition", true),
+					),
+				},
+				{
+					path: "/stack/test.tm",
+					add: generateFile(
+						labels("test"),
+						str("content", "test"),
+						boolean("condition", false),
+					),
+				},
+			},
+			want: []result{
+				{
+					name:      "test",
+					condition: false,
```

Comment:
hmmm a  change on the string representation of the file could change the ordering according to condition and break the test I think :thinking:.

I'm kinda torn on the sorting using String() + also torn if we should have ordering. We already re-order things on generate, on genfile/genhcl I dont think we have a strong constraint on ordering other than making the tests easier, but maybe we could write the tests in a way that is order independent. Or at least being more explicit on the ordering instead of using the string representation :thinking:

Reply:
you're right, there's no constraint other than making the tests easier (at this point) but also I don't have anything against making the order of generated files deterministically. We definitely need to talk about this, I did the simpler approach...

Reply:
Sure thing, I'm not a 100% sure on either approach (sorting or not, and if we sort which criteria), lets discuss in the future :rocket:

Reply:
Ah...actually I forgot we have some other maps involved in the parsing, so we would lose deterministic results. Since we replaced maps with slices my first instinct was that now we have deterministic output, just unordered, but that is not true here, so I would be pro the ordering because deterministic output is cool =P. Still wondering if the sorting implementation is the best, but that can be assesed/changed later.